runtime_scan/pkg/provider/azure/client.go

```diff
@@ -222,19 +223,48 @@ func processVirtualMachineListIntoAssetTypes(vmList armcompute.VirtualMachineLis
 	return ret, nil
 }
 
-func getVMInfoFromVirtualMachine(vm *armcompute.VirtualMachine) (models.AssetType, error) {
+type rootVolumeInfo struct {
```

Comment:
Since you define this both here and in aws, can we have a general type "rootVolume" defined in types?

Reply:
we can but it's just an auxiliary struct and azure doesn't have a device name. WDYT?

Reply:
I would avoid tying the provider implementations together with shared structs, IMO this is an implementation detail and it just happens to look/be named the same because the same developer wrote it, but they aren't really the same.

Reply:
Yeah but we are assuming there is a rootVolume for each provider, that has a size and encryption option, no? This is a general concept that exists also in the api on VMInfo, that is provider agnostic

Reply:
I think if @akpsgit takes my suggestion from above about making a rootVolume struct in the OAS we could reuse the models.RootVolume here potentially.

Reply:
the problem is that aws has a root device name but azure doesn't, that's why i didn't make it common. Not sure about using an API struct here, I think that each provided will have it's own fields and maybe not worth make it common. If you think that it's better to use the API, i will remove the root device name and use it from the instance

Reply:
yeah probably better to leave getVMInfoFromVirtualMachine to convert it to the API objects 👍